### PR TITLE
feat(flag): replace '--slow' with '--parallel'

### PR DESCRIPTION
**Pull request by knqyf263**
## Description
We now need to control the number of parallels. This PR deprecates `--slow` and adds `--parallel`.

### Before

```
$ trivy image --slow YOUR_IMAGE
```

### After

```
$ trivy image --parallel 1 YOUR_IMAGE
```

## Related issues
- Close https://github.com/aquasecurity/trivy/issues/5571

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [x] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

**Comment by knqyf263** [comment]
@Moniseeta @santhosh1729 Please take a look.

File: docs/docs/references/configuration/cli/trivy_sbom.md [hunk 1]
```diff
@@ -42,6 +42,7 @@ trivy sbom [flags] SBOM_PATH
       --no-progress                 suppress progress bar
       --offline-scan                do not issue API requests to identify dependencies
   -o, --output string               output file name
+      --parallel int                number of goroutines enabled for parallel scanning (default 5)
```
Comment (DmitriyLewen) [inline docs/docs/references/configuration/cli/trivy_sbom.md:45]: Looks like we don't use parallel for SBOM target.
Perhaps we can disable this flag.

File: pkg/flag/kubernetes_flags.go [hunk 2]
```diff
@@ -160,14 +149,7 @@ func (f *K8sFlagGroup) ToOptions() (K8sOptions, error) {
 	if err != nil {
 		return K8sOptions{}, err
 	}
-	var parallel int
-	if f.Parallel != nil {
-		parallel = getInt(f.Parallel)
-		// check parallel flag is a valid number between 1-20
-		if parallel < 1 || parallel > 20 {
```
Comment (DmitriyLewen) [inline pkg/flag/kubernetes_flags.go:167]: Perhaps we need to keep these (or other) restrictions for new scan flag:
https://github.com/aquasecurity/trivy/blob/c8eaa4c0f961c67a675d0d35d20171c1d58de257/pkg/flag/scan_flags.go#L62-L67

File: pkg/semaphore/semaphore.go [hunk 3]
```diff
@@ -4,27 +4,9 @@ import "golang.org/x/sync/semaphore"
 
 const defaultSize = 5
```
[the remark this answers is not in the document]
Comment (knqyf263) [reply pkg/semaphore/semaphore.go:5]: It's true in Trivy OSS, but Trivy Premium initializes the options themselves. We should take care of the case where `opt.Parallel` is 0.

File: pkg/commands/artifact/run.go [hunk 4]
```diff
@@ -637,7 +637,6 @@ func initScannerConfig(opts flag.Options, cacheClient cache.Cache) (ScannerConfi
 			SBOMSources:       opts.SBOMSources,
 			RekorURL:          opts.RekorURL,
 			//Platform:          opts.Platform,
-			Slow:         opts.Slow,
```
[the remark this answers is not in the document]
Comment (knqyf263) [reply pkg/commands/artifact/run.go:640]: I've added it in another PR and forgot to pick the change into this PR. I'll add it.
https://github.com/aquasecurity/trivy/pull/5180/files#diff-3b6c25b1435d1150f7492a6c5b47429c0f0bf1f630340582163cece9823168b7R631

[the remark this answers is not in the document]
Comment (knqyf263) [reply pkg/commands/artifact/run.go:640]: Done
https://github.com/aquasecurity/trivy/pull/5572/commits/3de5efd46a2a958230642052d0ea112f772f1793

File: docs/docs/references/configuration/cli/trivy_sbom.md (hunk 1 above)
Comment (knqyf263) [reply docs/docs/references/configuration/cli/trivy_sbom.md]: Nice catch. Done https://github.com/aquasecurity/trivy/pull/5572/commits/ca71aace803fd20f6115bdab58e762a4938a0b91

File: pkg/semaphore/semaphore.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (DmitriyLewen) [reply pkg/semaphore/semaphore.go:5]: I expected this
Thanks for clarifying this for me

File: pkg/flag/kubernetes_flags.go (hunk 2 above)
Comment (knqyf263) [reply pkg/flag/kubernetes_flags.go:167]: I give a special meaning to `--parallel 0`.
https://github.com/aquasecurity/trivy/pull/5572/commits/c08df01eb27b4f2189823175785b18233bc9c8dc

I'm not sure about the maximum. Recent computers can have many CPU cores and enough bandwidth.

Comment (DmitriyLewen) [reply pkg/flag/kubernetes_flags.go:167]: okay, good. 
Can you fix test?

Comment (knqyf263) [reply pkg/flag/kubernetes_flags.go:167]: fixed

**Review by DmitriyLewen — APPROVED** [review]
LGTM

**Pull request merged**